### DISCUSSION TITLE
DeploymentEnvironment cannot be extended via !ref

### DISCUSSION THREAD

**Discussion by AnnejanBarelds** [structurizr/cli#112] [Q&A] [answered]
It looks like a deploymentEnvironment element cannot be extended by using !ref. The CLI throws an error when I try it:

`StructurizrDslParserException: Unexpected tokens (expected: deploymentGroup, deploymentNode, ->)`

A minimal repro looks like this:

```
workspace {
    model {
        de = deploymentEnvironment "DeploymentEnvironment"

        !ref de {
            dn = deploymentNode "DeploymentNode"
        }
    }
}
```

Obviously the intend is to have the deploymentEnvironment in another file and !include that file.

Should this work, or am I misinterpreting something?

**Comment by simonbrowndotje** [comment, marked as answer]
Try this instead:

```
workspace {
    model {
        de = deploymentEnvironment "DeploymentEnvironment" {
        }

        !ref de {
            dn = deploymentNode "DeploymentNode"
        }
        
    }
}
```

**Reply by AnnejanBarelds** [reply]
So simple ;). Thanks!

Out of curiosity: any reason why the accolades are needed in this case? They're not required when referencing e.g. a deploymentNode.

**Reply by simonbrowndotje** [reply]
I would need to check, but I suspect the DSL parser thought you were inside the `deploymentEnvironment` block (hence the error message `expected: deploymentGroup, deploymentNode, ->)`), and `!ref` isn't permitted inside a deployment environment.

**Reply by simonbrowndotje** [reply]
My suspicion was correct, so I just added some code the DSL parser -> https://github.com/structurizr/dsl/commit/da9e4987b0a336cdf3e793caab53538df32213aa
